src/main/java/com/epam/training/olha_haichenkova/task_3/page/YopmailMainPage.java

```diff
@@ -53,14 +72,20 @@ private void closeAdvertisementIfAppear() {
     }
 
     private void closeAdvertisementPopUp() {
-        driver.switchTo().frame(YOP_OUTER_IFRAME_ID);
-        try{
-            closeAdvertisementButton.click();
-        } catch (NoSuchElementException e){
-            driver.switchTo().frame(YOP_INNER_IFRAME_ID);
-            closeAdvertisementButton.click();
+        if(!closeAdvertisementBanner.isEmpty()) {
+            new WebDriverWait(driver, ofMillis(5000))
+                    .pollingEvery(ofMillis(500))
+                    .until(frameToBeAvailableAndSwitchToIt(cssSelector(YOP_OUTER_IFRAME_CSS)));
```

Comment:
Please move this waiter to a separate method somewhere on the Abstract page and only provide the condition for it here, also we don't need to hardcode the time periods